### ISSUE TITLE
Flags mislinked for Qt5 while the corresponding Enum works

### ISSUE THREAD

**Issue by altendky** [Czaki/sphinx-qt-documentation#10]
`QtWidgets.QFileDialog.Options` gets an incorrect link when referencing the Qt5 documentation.  `Options` is a flags while `Option` is an enum, though they both seem to share the `#Option-enum` anchor.  If I switch to `QtWidgets.QFileDialog.Option` the link works.  I'm not really sure the difference or which I should be using though the flags sounds maybe a bit better.  For now I'll just go with the enum.

Actual: https://doc.qt.io/qt-5/qfiledialog-options.html
Correct: https://doc.qt.io/qt-5/qfiledialog.html#Option-enum

Is there a `.inv` URL for PySide2 or Qt5 itself?  Handling translation between unsynchronized systems seems like it could be a significant burden.

https://github.com/Czaki/sphinx-qt-documentation/blob/25d2dc15fb14a3b2b2ab45de889c2e0eabb76cd4/sphinx_qt_documentation/__init__.py#L131-L135

This is pretty messy and I didn't look for all the things it probably breaks but...  I'll share it for reference as it fixes this one thing.

```python
        if app.config.qt_documentation == "Qt5":
            html_name = uri.split("/")[-1]
            if name == 'class':
                match = re.search(r'(-([^\.]+)s)(?=\.)', html_name)
                if match:
                    html_name = html_name.replace(match[1], '') + '#' + match[2].title()
                    name = 'enum'
            uri = "https://doc.qt.io/qt-5/" + html_name
            if name == 'enum':
                uri += "-enum"
```

**Comment by Czaki** [comment]
> Is there a `.inv` URL for PySide2 or Qt5 itself? Handling translation between unsynchronized systems seems like it could be a significant burden.

I agree but I only found `.inv` file for PyQt5. 

I will check this case in next week and try to fix it in generic way.

**Comment by altendky** [comment]
Did you check with the pyside2 folks, either an issue report or on gitter/IRC? If not, I can do so.

**Comment by altendky** [comment]
This at least downloads...

https://doc.qt.io/qtforpython/objects.inv

I haven't done anything with it yet.

**Comment by altendky** [comment]
Welp, using just that `.inv` without `sphinx-qt-documentation` it does manage to link `` :class:`PySide2.QtCore.QObject` `` to https://doc.qt.io/qtforpython/PySide2/QtCore/QObject.html#PySide2.QtCore.QObject.  So it seems to work.  So at least for either Python docs `sphinx-qt-documentation` doesn't need to be guessing URLs.  Just dealing with adding the leading `PyQt5.` or `PySide2.` at the beginning (or whatever other pieces like that I haven't learned about yet).  So that's progress for the Python doc options.

I do wonder if I should rethink my preference but I always use the Qt5 C++ docs directly for the general classes.

`QEvent.Type` is a bit odd and the URL is as well.

https://doc.qt.io/qtforpython/PySide2/QtCore/QEvent.html?PySide2.QtCore.QEvent#PySide2.QtCore.PySide2.QtCore.QEvent.Type

![image](https://user-images.githubusercontent.com/543719/90064681-27063480-dcb9-11ea-87a9-f4646e1dff06.png)

`QEvent` itself is sensible.

![image](https://user-images.githubusercontent.com/543719/90064625-0f2eb080-dcb9-11ea-9c1d-da8e0c7da43f.png)

So it looks like there will be some PRs for PySide2 though it's easy enough to work around and search for the thing and also the thing with it's first two elements prepended as well...  `:]`

**Comment by Czaki** [comment]
I'm currently out of computer most of the day (till end of this week).
Thanks for finding .inv file for PySide2. 

Because of Quality of Documentation I Prefer to link to Qt5 doc. 

This project starts because PyQt5 use own namespace and I  cannot find other `.inv` file. But It starts long time ago, and I extract it to separate package after one SO question. So It think that it still will be nice to Add translation PySied2 `.inv` file to Qt5 C++ docs. 

I'm open on PR. I start also #11 to simplify adding tests. Maybe You could suggest how test such case?

**Comment by altendky** [comment]
Yeah, I definitely agree there's still value here.  I'm poking around and trying to get a hacky thing working.  Then we can throw it away and do something good.  `:]`  I'll plan to take a look at #11, thanks.

**Comment by Czaki** [comment]
The current link to pyside2 objects.inv is https://doc.qt.io/qtforpython-5/objects.inv
